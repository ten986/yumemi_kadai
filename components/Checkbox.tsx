import React, { ChangeEventHandler } from 'react'

type CheckBoxProps = {
  id: string
  value: string
  checked: boolean
  onChange: ChangeEventHandler<HTMLInputElement>
}

// チェックボックス
const CheckBox: React.FC<CheckBoxProps> = ({
  id,
  value,
  checked,
  onChange,
}) => {
  return (
    <>
      <input
        id={id}
        type="checkbox"
        name="checkbox"
        checked={checked}
        onChange={onChange}
        value={value}
      />
      <label htmlFor={id}>{value}</label>
    </>
  )
}

export default CheckBox
