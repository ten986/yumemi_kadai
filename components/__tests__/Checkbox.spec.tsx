import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  it('スナップショットが一致するか', () => {
    const mockFn = jest.fn()
    const props = {
      id: 'checkbox',
      value: '北海道',
      checked: false,
      onChange: mockFn,
    }
    const checkbox = shallow(<Checkbox {...props} />)

    // スナップショット
    expect(checkbox).toMatchSnapshot()
  })
  it('onChangeが呼ばれるか', () => {
    const mockFn = jest.fn()
    const props = {
      id: 'checkbox',
      value: '北海道',
      checked: false,
      onChange: mockFn,
    }
    const checkbox = shallow(<Checkbox {...props} />)

    // onChange が呼ばれるか
    checkbox.find('div').find('input').simulate('change')
    expect(mockFn.mock.calls.length).toEqual(1)
  })
  it('テキストがvalueと一致するか', () => {
    const mockFn = jest.fn()
    const props = {
      id: 'checkbox',
      value: '北海道',
      checked: false,
      onChange: mockFn,
    }
    const checkbox = shallow(<Checkbox {...props} />)

    // テキストが北海道になるか
    expect(checkbox.text()).toBe('北海道')
  })
})
