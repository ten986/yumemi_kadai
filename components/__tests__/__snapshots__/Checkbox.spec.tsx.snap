// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox スナップショットが一致するか 1`] = `
<div
  className="wrapper"
>
  <input
    checked={false}
    id="checkbox"
    name="checkbox"
    onChange={[MockFunction]}
    type="checkbox"
    value="北海道"
  />
  <label
    htmlFor="checkbox"
  >
    北海道
  </label>
</div>
`;
